app/src/main/java/com/a1ishka/bankaccount/presentation/transaction/TransactionViewModel.kt

```diff
@@ -4,19 +4,30 @@ import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.a1ishka.bankaccount.data.repository.TransactionRepositoryImpl
 import com.a1ishka.bankaccount.domain.Transaction
+import com.a1ishka.bankaccount.domain.repository.TransactionRepository
 import com.a1ishka.bankaccount.util.toTransactionEntity
+import com.a1ishka.bankaccount.util.toTransactionList
+import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
-
+@HiltViewModel
 class TransactionViewModel @Inject constructor(
-    private val transactionRepositoryImpl: TransactionRepositoryImpl
+    private val transactionRepository: TransactionRepository
 ) : ViewModel() {
     private val _state = MutableStateFlow(TransactionState())
     val transactionState: MutableStateFlow<TransactionState> = _state
```

Comment:
must be StateFlow type instead of MutableStateFlow there. This method (when we have for example private _state - mutable and public state - immutable) called private setter and need to avoid state changes directly through the property